src/index.js

```diff
@@ -15,14 +16,15 @@ ReactDOM.render(
   <React.StrictMode>
   <BrowserRouter>
     <ProductProvider>
-      <CartProvider>
-        <WishlistProvider>
-          <App />
-        </WishlistProvider>
-      </CartProvider>
+      <AuthProvider>
+        <CartProvider>
+          <WishlistProvider>
+            <App />
+          </WishlistProvider>
+        </CartProvider>
+      </AuthProvider>
     </ProductProvider>
-  </BrowserRouter>
-   
+  </BrowserRouter> 
   </React.StrictMode>,
   document.getElementById("root")
```

Comment:
Suggestion you could abstract out all the providers into one provider file.

Example: 


<HifiStoreProvider>
           <App />
</HigiStoreProvider>